### PR TITLE
Ratings API & catalog-manager Operator

### PR DESCRIPTION
**Pull request by aleixhub**

### CODE REVIEW

File: ratings/api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM registry.access.redhat.com/ubi8/python-38:latest
```
Comment (jkupferer) [inline ratings/api/Dockerfile:1]: Should probably use python-39

File: ratings/api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+COPY . /tmp/src
+
+RUN chown -R 1001:0 /tmp/src && \
+    chmod -R g+w /tmp/src
+
+USER 1001
+
+RUN /usr/libexec/s2i/assemble
+
+EXPOSE 80
```
Comment (jkupferer) [inline ratings/api/Dockerfile:14]: Expose port 80? I would have expected a high port?

File: ratings/api/app.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+CREATE_RATINGS_TABLE = """CREATE TABLE IF NOT EXISTS ratings (
+                    provision_uuid varchar(64) NOT NULL, email varchar NOT NULL, 
+                        catalog_item_id int NOT NULL, rating int, comment TEXT, 
+                        created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+                        updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+                        PRIMARY KEY(provision_uuid, email),
+                        FOREIGN KEY(provision_uuid) REFERENCES provisions(uuid) ON DELETE CASCADE,
+                        FOREIGN KEY(email) REFERENCES students(email) ON DELETE CASCADE,
+                        FOREIGN KEY(catalog_item_id) REFERENCES catalog_items(id) ON DELETE CASCADE
+                    );"""
```
Comment (jkupferer) [inline ratings/api/app.py:21]: Hrm... It seems a bit odd to me having the application manage the database schema. I would expect this sort of thing to be elsewhere? As a database administrator I wouldn't want applications creating tables or even having permissions to createt tables.

File: ratings/operator/operator/operator.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+#----------------------------------------------------------------------------#
+# Launch.
+#----------------------------------------------------------------------------#
+@kopf.on.startup()
+async def on_startup(settings: kopf.OperatorSettings, logger, **_):
+    logger.info("Booting up")
+    event_loop = asyncio.get_event_loop()
+    event_loop.create_task(every(300, set_ratings, logger))
+    event_loop.run_forever()
```
Comment (jkupferer) [inline ratings/operator/operator/operator.py:96]: This does not appear to be actually using kopf? No need to bring kopf in for this?

File: ratings/api/app.py (hunk 3 above)
Comment (aleixhub) [reply ratings/api/app.py:24]: IMHO there are benefits in having this logic here, for example, if a new version of the application needs a table with the current data then you need to create this table at the specific moment of deploying the new version as you will need the data from that instant.
Also having encapsulated this logic in the application gives the developer a better understanding of the model.
What I don't like is sharing this DB between two services as it contains foreign keys related to other tables, but I guess is a compromise we need to take (as having a new database will mean keeping in sync those tables. And that is a problem that worth to be avoided if possible)

Do you see this as a vulnerability risk?

**Pull request merged**